lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -771,7 +771,7 @@ func (uw *urlWatcher) watchForUpdates() {
 		err = uw.readObjectUpdateStream(resp.Body)
 		_ = resp.Body.Close()
 		if err != nil {
-			if !errors.Is(err, io.EOF) {
+			if !(errors.Is(err, io.EOF) || errors.Is(err, context.Canceled)) {
```

Comment:
@zekker6 I think it should be `if !errors.Is(err, io.EOF) && !errors.Is(err, context.Canceled)`